Source/WebCore/dom/Document.cpp

```diff
@@ -9019,20 +9025,51 @@ HTMLElement* Document::topmostAutoPopover() const
 // https://html.spec.whatwg.org/#hide-all-popovers-until
 void Document::hideAllPopoversUntil(Element* endpoint, FocusPreviousElement focusPreviousElement, FireEvents fireEvents)
 {
-    Vector<Ref<HTMLElement>> popoversToHide;
-    for (auto& item : makeReversedRange(m_topLayerElements)) {
-        if (!is<HTMLElement>(item))
-            continue;
-        auto& element = downcast<HTMLElement>(item.get());
-        if (element.popoverState() == PopoverState::Auto) {
-            if (&element == endpoint)
-                break;
-            popoversToHide.append(element);
+    auto closeAllOpenPopovers = [&]() {
+        while (RefPtr popover = topmostAutoPopover())
+            popover->hidePopoverInternal(focusPreviousElement, fireEvents);
+    };
+    if (!endpoint)
+        return closeAllOpenPopovers();
+    auto autoPopoverList = [&]()  {
+        Vector<Ref<HTMLElement>> popovers;
+        for (auto& item : m_topLayerElements) {
+#if ENABLE(FULLSCREEN_API)
+            if (item->hasFullscreenFlag())
+                continue;
+#endif
+            if (auto* candidate = dynamicDowncast<HTMLElement>(item.get()); candidate && candidate->popoverState() == PopoverState::Auto) {
+                if (!is<HTMLDialogElement>(candidate) || !downcast<HTMLDialogElement>(candidate)->isOpen())
```

Comment:
I don't think we need all of these checks, quoting the spec:

> The [Document ](https://whatpr.org/html/9198/3d8fca2...cb6b5f6/dom.html#document)has an auto popover list , which is a [list ](https://infra.spec.whatwg.org/#list)of all the elements in the [Document ](https://whatpr.org/html/9198/3d8fca2...cb6b5f6/dom.html#document)'s [top layer ](https://fullscreen.spec.whatwg.org/#top-layer)whose [popover ](https://whatpr.org/html/9198/3d8fca2...cb6b5f6/popover.html#attr-popover)attribute is in the [auto ](https://whatpr.org/html/9198/3d8fca2...cb6b5f6/popover.html#attr-popover-auto-state)state, in the same order.

Reply:
This is what chromium and my previous patch do implicitly by the way the autopopover vector is maintained. Not doing these checks cause failures and even crashes.

Reply:
Ah, I guess you meant the checks in the lambda, not in general. Yes looks like we do not need them in the lambda, removed.